src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Refactoring/RenameEndpoint.cs

```diff
@@ -68,17 +71,42 @@ internal class RenameEndpoint : IRenameEndpoint
             return new RegistrationExtensionResult(ServerCapability, options);
         }
 
-        public async Task<WorkspaceEdit?> Handle(RenameParamsBridge request, CancellationToken cancellationToken)
+        /// <inheritdoc/>
+        protected override string EndpointName => LanguageServerConstants.RazorRenameEndpointName;
+
+        /// <inheritdoc/>
+        protected override Task<WorkspaceEdit?> HandleWithoutSingleServerAsync(RenameParamsBridge request, DocumentContext documentContext, RazorCodeDocument codeDocument, SourceText sourceText, CancellationToken cancellationToken)
+            => Task.FromResult<WorkspaceEdit?>(null);
```

Comment:
Ah I see, so if we override `Handle` then we basically don't use this. I wonder if we should do away with the `HandleWithoutSingleServeAsync` and continue to rely on registering different services based on the flag to keep things simple

Reply:
I removed `HandleWithoutSingleServerAsync` and just moved to only `HandleInRazorAsync` if there is special logic there. Otherwise I agree we can fallback to cshtml server. It's starting to click a little better now